src/blocks/Text/DefaultTextBlockEditor.jsx

```diff
@@ -188,6 +191,81 @@ export const DefaultTextBlockEditor = (props) => {
     }
   }, [onSelectBlock, selected, block]);
 
+  const slashCommand = data.plaintext?.trim().match(/^\/([a-z]*)$/);
+
+  const useAllowedBlocks = !isEmpty(allowedBlocks);
+
+  const filteredBlocksConfig = filter(blocksConfig, (item) => {
+    if (useAllowedBlocks) {
+      return allowedBlocks.includes(item.id);
+    } else {
+      return typeof item.restricted === 'function'
+        ? !item.restricted({ properties, block: item })
+        : !item.restricted;
+    }
+  }).sort((a, b) => (a.title < b.title ? -1 : 1));
+
+  // Remove non matching
+  if (slashCommand) {
+    remove(
+      filteredBlocksConfig,
+      (block) => block.title.toLowerCase().indexOf(slashCommand[1]) === -1,
+    );
+  }
+
+  const slashMenuSize = keys(filteredBlocksConfig).length;
+  if (slashMenuSelected > slashMenuSize - 1) {
+    setSlashMenuSelected(slashMenuSize - 1);
+  }
+
+  const onKeyDown = ({ editor, event }) => {
+    if (slashCommand) {
```

Comment:
This should be a separate plugin, as a part of the https://github.com/eea/volto-slate/blob/0ed38ac64cd269f8c550598932fac0a28d939f09/src/blocks/Text/keyboard/index.js#L23

Keyboard handlers are handled by priority, and returning True from a handler means event is handled, so no more "propagation". This also keeps the implementation clean, SlateEditor is just an editor while the Text block has integration, such as the slash menu.